xml/System/DateTime.xml

```diff
@@ -2033,7 +2033,7 @@ For applications in which portability of date and time data or a limited degree
 
  The fractional part of `value` is the fractional part of a day. For example, 4.5 is equivalent to 4 days, 12 hours, 0 minutes, 0 seconds, 0 milliseconds, and 0 ticks.
 
- The `value` parameter is rounded to the nearest millisecond.
+ In .NET 6 and earlier versions only, the `value` parameter is rounded to the nearest millisecond. In .NET 7 and and later versions, the <xref:System.Double> precision of the `value` parameter is used to round to the nearest tick.
```

Comment:
Nearest tick isn't really accurate.

Inaccuracies can creep in due to `double` being a `binary floating-point value`. For example, the `double` value `9.2` is actually representing `9.199999999999999289457264239899814128875732421875` and so `AddHours(9.2)` will end up computing a tick count using this actual represented value.

Some precision of this "actual represented value" will get lost since `double` can only accurately represent whole integers up to `2^53`.

As I explained in https://github.com/dotnet/dotnet-api-docs/issues/8292#issuecomment-1211385532, this ultimately only guarantees `millisecond` accuracy still. We may include additional precision in the computed result and there may be non zero data for sub millisecond ranges. However, it is not guaranteed to be "exact" for such scenarios and so we cannot guarantee "microsecond", "tick", or "nanosecond" level accuracy.

Reply:
@tannergooding do you suggest not changing the doc then? or we need to write it some other way?

Reply:
There is probably some good way to describe this more succinctly, but I'm not sure quite how we'd do that atm as understanding the limitations here require more in depth technical explanations.

Maybe we could describe this as in some terminology to describe that in .NET 6 it was `Millisecond/Millisecond` but in .NET 7 it is `Millisecond/Nanosecond`.

That is, in .NET 6 we explicitly rounded everything to the nearest millisecond and dropped any additional precision.

In .NET 7, we no longer explicitly round and so we can have nanosecond level precision, but may still only persist user inputs to the millisecond level due to how binary-floating point numbers work.

Reply:
Anything that would mention nanosecond as the precision would be extremely wrong, as DateTime is only storing the value as a number of ticks, and 1 tick = 100 ns. I don't understand what's wrong in what I've writen: the code uses the double precision (as much as it can , and yes, double can be inaccurate), and the result in expressed in ticks. In .NET 7, we *also* drop any additional precision after the tick, and that's why I suggest that we should *round* to the nearest tick.